src/docmanager/index.ts

```diff
@@ -399,7 +442,10 @@ class DocumentManager {
   }
 
   private _data: { [key: string]: Private.IDocumentData } = Object.create(null);
-  private _fileTypes: { [key: string ]: string[] } = Object.create(null);
+  private _modelFactories: { [key: string]: IModelFactory } = Object.create(null);
```

Comment:
It could be that I missed something obvious, so I could be convinced to try them again.